src/scandipwa/app/component/GoogleTagManager/events/AddToCart.event.js

```diff
@@ -24,6 +26,8 @@ class AddToCartEvent extends BaseEvent {
      * Bind add to cart
      */
     bindEvent() {
+        if (!isEventEnabled(EVENT_ADD_TO_CART)) return;
```

Comment:
I think it's worth moving all these checks on event activation to the `Event.observer` function itself to avoid code duplication.